consensus/types/src/blob.rs

```diff
@@ -0,0 +1,223 @@
+use crate::EthSpec;
+use arbitrary::{Arbitrary, Unstructured};
+use kzg::{BlobTrait, KzgPreset};
+use rand::Rng;
+use safe_arith::SafeArith;
+use serde::{
+    de::{self, Visitor},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+use ssz::{Decode, DecodeError, Encode};
+use std::hash::{Hash, Hasher};
+use std::sync::Arc;
+use tree_hash::{PackedEncoding, TreeHash};
+
+#[derive(Clone, PartialEq, Eq)]
+pub struct SigpBlob<E: EthSpec>(Arc<<E::Kzg as KzgPreset>::Blob>);
+
+impl<E: EthSpec> SigpBlob<E> {
```

Comment:
I think we should keep the original `Blob` name rather than `SigpBlob`. It matches the spec and might be nicer for other users of our consensus types. We don't seem to use the c-kzg blob type anywhere in lighthouse directly either (we don't import it anywhere at least).

Reply:
<img width="118" alt="image" src="https://github.com/sigp/lighthouse/assets/742762/279f8357-738b-4f1f-bff9-954898399a2d">

Reply:
I've renamed this to `WrappedBlob`